python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -78,6 +79,38 @@ def run_loop():
         self.udf_map_asyncio_thread = thread
 
 
+def plan_project_op(
+    op: Project, physical_children: List[PhysicalOperator]
+) -> MapOperator:
+    assert len(physical_children) == 1
+    input_physical_dag = physical_children[0]
+
+    columns = op.cols
+
+    def fn(batch: "pa.Table") -> "pa.Table":
+        try:
+            return batch.select(columns)
+        except Exception as e:
+            _handle_debugger_exception(e)
```

Comment:
This will break for Pandas, right? 

We'd add support for it here (until we drop it completely as an internal block format)

Reply:
We force the batch format to be pyarrow in above -> https://github.com/ray-project/ray/pull/48635/files#diff-7aaeab5974c738dac5f137b1ba54dbb32d9fac4cfa200fe7e429ec1294863e72R251

Reply:
what's the use case you think that may fail?

Reply:
This is a batch format, so it shouldn't matter what the internal block format is, right?